pkg/updater/repository/link.go

```diff
@@ -0,0 +1,38 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
+
+package repository
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/google/renameio/v2"
+)
+
+func linkRead(linkPath string) (string, error) {
+	return filepath.EvalSymlinks(linkPath)
+}
+
+func linkExists(linkPath string) (bool, error) {
+	_, err := os.Stat(linkPath)
+	if errors.Is(err, os.ErrNotExist) {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func linkSet(linkPath string, targetPath string) error {
+	return renameio.Symlink(targetPath, linkPath)
```

Comment:
Is it worth pulling a new dependency for just a single function? 

https://github.com/google/renameio/blob/v1.0.1/tempfile.go#L156-L187

Should we just copy the function code into this file?

Reply:
The content of the function are quite tricky. I can copy it if you feel strongly about it but I think this also means I'd still have to bring their Apache 2 license somehow if I just copy/paste it

Reply:
+1 on what the reasoning fwk suggested by Gustavo.
Would love to get the answer to the questions suggested.

Reply:
That's fine, I don't mind importing if that's the preference. I was worried this would entail finding a new way to import the licence since Apache 2 requires keeping the copyright but if that's not expected all good, I'll vendor the function